ambry-network/src/main/java/com.github.ambry.network/NetworkClient.java

```diff
@@ -93,6 +93,7 @@ public NetworkClient(Selector selector, NetworkConfig networkConfig, NetworkMetr
   public List<ResponseInfo> sendAndPoll(List<RequestInfo> requestInfos)
       throws IOException {
     long startTime = time.milliseconds();
+    logger.trace("Initiating Send and poll at " + startTime);
```

Comment:
And i'm not sure how you would debug based on this trace logging. If *some* cycles are taking more time, that shows up in the 95th and 99th percentile latencies of the histogram of send and poll anyway  - how does this additional trace logging help?

Generally trace logging is used to track events and to make sure they happened - I don't think times are very useful.